scripts/docker-integration-tests/query_fanout/warning.sh

```diff
@@ -0,0 +1,309 @@
+  EXPECTED_HEADER=$2
+  trap clean_headers EXIT
+
+   curl -sSL -D headers -H "M3-Limit-Max-Series:$LIMIT" \
+    "http://0.0.0.0:7201/api/v1/label/val/values"
+  ACTUAL_HEADER=$(cat headers | grep M3-Results-Limited | cut -d' ' -f2 | tr -d "\r\n")
+  test $ACTUAL_HEADER = $EXPECTED_HEADER
+}
+
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff test_label_values 100
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff test_label_values 1 max_fetch_series_limit_applied
+
+
+GRAPHITE="foo.bar.$t"
+function write_carbon {
+  CLUSTER=$1
+  case $CLUSTER in
+    coordinator-cluster-a )
+      PORT=7204 ;;
+    coordinator-cluster-b )
+      PORT=17204 ;;
+    coordinator-cluster-c )
+      PORT=27204 ;;
+    *)
+      echo $CLUSTER "is not a valid coordinator cluster"
+      exit 1
+  esac
+
+  NUM=$2
+  echo "Writing $NUM metrics to $1 [0.0.0.0:$PORT]"
+  set +x
+  for (( i=0; i<$NUM; i++ ))
+  do
+    echo "$GRAPHITE.$i.$CLUSTER 1 $t" | nc 0.0.0.0 $PORT
+  done
+  set -x
+}
+
+function render_carbon {
+  LIMIT=$1
+  EXPECTED=$2
+  EXPECTED_HEADER=$3
+  trap clean_headers EXIT
+
+  start=$(($t))
+  end=$(($start+200))
+  RESPONSE=$(curl -sSL -D headers -H "M3-Limit-Max-Series:$LIMIT" \
+    "http://localhost:7201/api/v1/graphite/render?target=countSeries($GRAPHITE.*.*)&from=$start&until=$end")
+  ACTUAL=$(echo $RESPONSE | jq .[0].datapoints[0][0])
+  ACTUAL_HEADER=$(cat headers | grep M3-Results-Limited | cut -d' ' -f2 | tr -d "\r\n")
+  test $ACTUAL = $EXPECTED && test $ACTUAL_HEADER = $EXPECTED_HEADER
+}
+
+function find_carbon {
+  LIMIT=$1
+  EXPECTED_HEADER=$2
+  trap clean_headers EXIT
+
+  RESPONSE=$(curl -sSL -D headers -H "M3-Limit-Max-Series:$LIMIT" \
+    "http://localhost:7201/api/v1/graphite/metrics/find?query=$GRAPHITE.*")
+  ACTUAL_HEADER=$(cat headers | grep M3-Results-Limited | cut -d' ' -f2 | tr -d "\r\n")
+  test $ACTUAL_HEADER = $EXPECTED_HEADER
+}
+
+# write 5 metrics to cluster a
+write_carbon coordinator-cluster-a 5
+# unlimited query against cluster a has no header
+ATTEMPTS=8 TIMEOUT=1 retry_with_backoff render_carbon 6 5
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff find_carbon 6
+# limited query against cluster a has header
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff render_carbon 4 4 max_fetch_series_limit_applied
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff find_carbon 4 max_fetch_series_limit_applied
+
+# Update write time as it will otherwise not be written correctly.
+t=$(date +%s)
+# write 10 metrics to cluster b
+write_carbon coordinator-cluster-b 10
+# unlimited query against cluster a has no header
+ATTEMPTS=8 TIMEOUT=1 retry_with_backoff render_carbon 16 15
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff find_carbon 16
+# remote limited query against cluster a has header
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff render_carbon 9 14 max_fetch_series_limit_applied
+ATTEMPTS=3 TIMEOUT=1 retry_with_backoff find_carbon 9 max_fetch_series_limit_applied
+
+
+echo "turning off one of the zones"
+docker-compose -f ${COMPOSE_FILE} stop coordinator-cluster-c
```

Comment:
Nice, awesome that this gets tested 👍 

Later we can also do a test to make sure queries "fail fast" when there's a slow zone too.  Probably need to rely on the GRPC load balancer strategy being able to support this mode (definitely would be annoying to have to bake this logic ourselves).